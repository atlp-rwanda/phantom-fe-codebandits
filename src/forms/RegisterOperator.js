import React, { useState, useEffect } from 'react';
import validator from 'validator';
import Button from '../components/Button.js';
import buses from '../images/buses.jpg';
import Input from '../components/Input.js';
import busMap from '../images/busMap.png';
import { Footer, Header } from '../Dummy.js';

function RegisterOperator() {
  const initialValues = {
    FirstName: '',
    LastName: '',
    Email: '',
    MobileNumber: '',
    Address: '',
    NationalId: ''
  };
  const [formValues, setFormValues] = useState(initialValues);
  const [formErrors, setFormErros] = useState({});
  const [isSubmit, setIsSubmit] = useState(false);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormValues({ ...formValues, [name]: value });
    console.log(formValues);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    setFormErros(validateForm(formValues));
    setIsSubmit(true);
  };

  useEffect(() => {
    if (Object.keys(formErrors).length === 0 && isSubmit) {
      console.log(formValues);
    }
  }, [formErrors]);

  const validateForm = (values) => {
    const errors = {};

    const validateEmail = (value) => {
      if (!values.Email) {
        errors.Email = 'email is required';
      } else if (!validator.isEmail(value)) {
        errors.Email = errors.Email = 'Wrong Email';
      }
    };
    validateEmail(values.Email);

    const namePattern = /^([a-zA-Z]+\s)*[a-zA-Z]+$/;

    if (!values.FirstName) {
      errors.FirstName = 'first name is required';
    } else if (!values.FirstName.match(namePattern)) {
      errors.FirstName = 'Enter a valid name';
    }

    if (!values.LastName) {
      errors.LastName = 'last name is required';
    } else if (!values.LastName.match(namePattern)) {
      errors.LastName = 'enter valid name';
    }

    const phoneRegex = /(0|7|8)\d{9}$/;
    if (!values.MobileNumber) {
      errors.MobileNumber = 'mobile number is required';
    } else if (!values.MobileNumber.match(phoneRegex)) {
      errors.MobileNumber = 'Enter valid phone number';
    }

    if (!values.Address) {
      errors.Address = 'address is required';
    }

    const IdRegex = /^[1-3](19|20)\d{2}[7-8]\d{7}[0-9]\d{2}$/;
    if (!values.NationalId) {
      errors.NationalId = 'Id is required';
    } else if (!values.NationalId.match(IdRegex)) {
      errors.NationalId = 'nationa Id must be 16 numbers and valid';
    }

    return errors;
  };

  return (
    <div>
      <Header />
      <div className="font-sans antialiased bg-grey-lightest">
        <div className="w-full bg-grey-lightest ">
          <div className="container mx-auto py-8 grid grid-cols-2">
            <div className="border">
              <img
                src={busMap}
                alt=" Bus IMG"
                className="w-full h-full object-cover"
              />
            </div>
            <div className="w-5/6 lg:w-2/2 mx-auto bg-white rounded shadow p-8">
              <div className="py-4 px-8 text-black text-xl font-bold">
                Register Operator
              </div>
              <form onSubmit={handleSubmit}>
                <div className="py-4 px-8">
                  <div className="flex mb-4">
                    <div className="w-1/2 mr-1 flex flex-col">
                      <Input
                        type="text"
                        name="FirstName"
                        labelName="First Name"
                        placeholder="Enter operator first name"
                        id="firstName"
                        value={formValues.FirstName}
                        onChange={handleChange}
                      />
                      <p className="text-red-800">{formErrors.FirstName}</p>
                    </div>

                    <div className="w-1/2 mr-1 flex flex-col">
                      <Input
                        type="text"
                        name="LastName"
                        labelName="Last Name"
                        placeholder="Enter operator last name"
                        id="lastName"
                        value={formValues.LastName}
                        onChange={handleChange}
                      />
                      <p className="text-red-800">{formErrors.LastName}</p>
                    </div>
                  </div>

                  <div className="flex mb-4">
                    <div className="w-1/2 mr-1 flex flex-col">
                      <Input
                        type="text"
                        name="Email"
                        labelName="Email"
                        placeholder="Enter operator email"
                        id="email"
                        value={formValues.Email}
                        onChange={handleChange}
                      />
                      <p className="text-red-800">{formErrors.Email}</p>
                    </div>

                    <div className="w-1/2 mr-1 flex flex-col ">
                      <Input
                        type="text"
                        name="MobileNumber"
                        labelName="Mobile Number"
                        placeholder="Enter operator mobile number"
                        id="mobileNumber"
                        value={formValues.MobileNumber}
                        onChange={handleChange}
                      />
                      <p className="text-red-800">{formErrors.MobileNumber}</p>
                    </div>
                  </div>

                  <div className="flex mb-4">
                    <div className="w-1/2 mr-1">
                      <label
                        htmlFor="company"
                        name="Company"
                        className="block text-grey-darker text-sm font-bold mb-2"
                      >
                        {' '}
                        company
                      </label>
                      <select
                        id="company"
                        placeholder="Select operator company"
                        className="appearance-none border rounded w-full py-2 px-3 text-grey-darker bg-gray-200"
                      >
                        <option className="text-grey-darker">
                          Select company
                        </option>
                        <option value="">Royal Express Limited </option>
                        <option value=""> City Express Limited </option>
                        <option value="">KIGALI BUS SERVICES</option>
                      </select>
                    </div>
                    <div className="w-1/2 mr-1 flex flex-col">
                      <Input
                        type="text"
                        name="Address"
                        labelName="Address"
                        placeholder="Enter operator working adress"
                        id="address"
                        value={formValues.Address}
                        onChange={handleChange}
                      />
                      <p className="text-red-800">{formErrors.Address}</p>
                    </div>
                  </div>

                  <div className="mb-4">
                    <div className="w-full mr-1 flex flex-col">
                      <Input
                        type="text"
                        name="NationalId"
                        labelName="National Id"
                        placeholder="Enter operator national ID number"
                        id="nationalId"
                        onChange={handleChange}
                      />
                      <p className="text-red-800">{formErrors.NationalId}</p>
                    </div>
                  </div>
                  <div className="flex justify-center align">
                    <Button name="Register Operator" type="submit" />
                  </div>
                </div>
              </form>
            </div>
          </div>
        </div>
      </div>
      <Footer />
    </div>
  );
}

export default RegisterOperator;
