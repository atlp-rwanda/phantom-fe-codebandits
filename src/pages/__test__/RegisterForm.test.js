import { shallow } from 'enzyme';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import RegisterDriver from '../../forms/RegisterDrivers.js';
import RegisterOperator, {
  LabelComponent
} from '../../forms/RegisterOperator.js';

describe('Registration', () => {
  describe('Operators', () => {
    it('Should render the operator form', () => {
      const elem = renderer
        .create(
          <BrowserRouter>
            <RegisterOperator />
          </BrowserRouter>
        )
        .toJSON();
      expect(elem).toMatchSnapshot();
    });
    it('Should render the driver form', () => {
      const elem = renderer
        .create(
          <BrowserRouter>
            <RegisterDriver />
          </BrowserRouter>
        )
        .toJSON();
      expect(elem).toMatchSnapshot();
    });
  });
  describe('Component', () => {
    const wrapper = shallow(
      <LabelComponent htmlFor={'test'} name={'test'}></LabelComponent>
    );
    it('Should return test', () => {
      expect(wrapper.text()).toEqual('test');
    });
  });
});
