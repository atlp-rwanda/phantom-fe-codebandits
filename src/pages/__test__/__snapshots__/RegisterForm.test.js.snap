// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Operators Should render the driver form 1`] = `
<div>
  <div
    className="font-sans antialiased bg-grey-lightest"
  >
    <div
      className="w-full bg-grey-lightest "
    >
      <div
        className="py-8  md:flex"
      >
        <div
          className="mr-2 hidden lg:block md:w-1/2"
        >
          <img
            alt=" Bus IMG"
            className=" h-50  w-full max-w-lg md:w-70 md:w-full lg:mx-auto lg:h-full"
            src={null}
          />
        </div>
        <div
          className="block bg-white rounded  p-2 w-full lg:max-w-3xl shadow-lg min-h-[70vh] "
        >
          <div
            className="py-4 px-8 text-black font-bold lg:px-4 md:px-4"
          >
            <h1
              className="text-center  text-2xl  font-rale font-bold md:ml-0 md:text-left"
            >
              Register Driver
            </h1>
          </div>
          <form
            onSubmit={[Function]}
          >
            <div
              className="py-4 px-4  w-full  lg:mx-auto "
            >
              <div
                className="block mb-4 xl:flex md:grid md:grid-cols-2 md:gap-2"
              >
                <div
                  className="mr-1 xl:w-1/2 "
                >
                  <div
                    className="mb-2 block font-raleway"
                  >
                    <label
                      className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                      htmlFor="firstname"
                    >
                      First Name
                    </label>
                    <input
                      className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      id="firstName"
                      name="FirstName"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Ex: Yves"
                      type="text"
                    />
                  </div>
                  <p
                    className="text-red-800"
                  />
                </div>
                <div
                  className="ml-1 xl:w-1/2"
                >
                  <div
                    className="mb-2 block font-raleway"
                  >
                    <label
                      className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                      htmlFor="lastname"
                    >
                      Last Name
                    </label>
                  </div>
                  <input
                    className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    id="lastName"
                    name="LastName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Hakuzimana"
                    type="text"
                  />
                  <p
                    className="text-red-800"
                  />
                </div>
              </div>
              <div
                className="block mb-4 xl:flex md:grid md:grid-cols-2  md:gap-2"
              >
                <div
                  className="mr-1 xl:w-1/2"
                >
                  <div
                    className="mb-2 block font-raleway"
                  >
                    <label
                      className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                      htmlFor="email"
                    >
                      Email
                    </label>
                    <input
                      className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      id="email"
                      name="email"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Ex: example@gmail.com"
                      type="email"
                    />
                  </div>
                  <p
                    className="text-red-800"
                  />
                </div>
                <div
                  className="ml-1 xl:w-1/2"
                >
                  <div
                    className="mb-2 block font-raleway"
                  >
                    <label
                      className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                      htmlFor="mobileNumber"
                    >
                      Mobile Number
                    </label>
                    <input
                      className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      id="mobileNumber"
                      name="mobileNumber"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="0789000000"
                      type="text"
                    />
                  </div>
                  <p
                    className="text-red-800"
                  />
                </div>
              </div>
              <div
                className="block mb-4 xl:flex md:grid md:grid-cols-2  md:gap-2"
              >
                <div
                  className="mr-1 xl:w-1/2"
                >
                  <label
                    className="block text-grey-darker text-sm font-bold mb-2"
                    htmlFor="company"
                    name="Company"
                  >
                    Company
                  </label>
                  <select
                    className="appearance-none border rounded w-full py-2 px-3 text-grey-darker bg-gray-200"
                    id="company"
                    placeholder="Select operator company"
                    required={true}
                  >
                    <option
                      className="text-grey-darker"
                      hidden={true}
                    >
                      Select company
                    </option>
                    <option
                      value=""
                    >
                      Royal Express Limited 
                    </option>
                    <option
                      value=""
                    >
                       City Express Limited 
                    </option>
                    <option
                      value=""
                    >
                      KIGALI BUS SERVICES
                    </option>
                  </select>
                </div>
                <div
                  className="ml-1 xl:w-1/2 "
                >
                  <div
                    className="mb-2 block font-raleway"
                  >
                    <label
                      className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                      htmlFor="address"
                    >
                      Address
                    </label>
                    <input
                      className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      id="address"
                      name="address"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Nyamirambo, Kigali"
                      type="text"
                    />
                  </div>
                  <p
                    className="text-red-800"
                  />
                </div>
              </div>
              <div
                className="block mb-4 xl:flex md:grid md:grid-cols-2 md:gap-2"
              >
                <div
                  className="mr-1 xl:w-1/2"
                >
                  <label
                    className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                    htmlFor="nationalId"
                  >
                    National ID
                  </label>
                  <input
                    className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    id="nationalId"
                    name="nationalid"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="1 1997 80020002 1 20"
                    type="text"
                  />
                  <p
                    className="text-red-800"
                  >
                     
                  </p>
                </div>
                <div
                  className=" ml-1 xl:w-1/2 "
                >
                  <label
                    className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                    htmlFor="license"
                  >
                    Driver License
                  </label>
                  <input
                    className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    id="license"
                    name="license"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Enter enter driver license number"
                    type="text"
                  />
                  <p
                    className="text-red-800"
                  />
                </div>
              </div>
              <div
                className="flex justify-center align"
              >
                <div>
                  <button
                    className="undefined bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
                    type="submit"
                  >
                    Register Driver
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Registration Operators Should render the operator form 1`] = `
<div>
  <div
    className="font-sans antialiased bg-grey-lightest"
  >
    <div
      className="w-full bg-grey-lightest "
    >
      <div
        className="py-8  md:flex"
      >
        <div
          className="mr-2 hidden lg:block md:w-1/2"
        >
          <img
            alt=" Bus IMG"
            className=" h-50  w-full max-w-lg md:w-70 md:w-full lg:mx-auto lg:h-full"
            src={null}
          />
        </div>
        <div
          className="block bg-white rounded  p-2 w-full lg:max-w-3xl shadow-lg min-h-[70vh] "
        >
          <div
            className="py-4 px-8 text-black font-bold lg:px-4 md:px-4"
          >
            <h1
              className="text-center  text-2xl  font-rale font-bold md:ml-0 md:text-left"
            >
              Register Operator
            </h1>
          </div>
          <form
            onSubmit={[Function]}
          >
            <div
              className="py-4 px-4  w-full  lg:mx-auto "
            >
              <div
                className="block mb-4 xl:flex md:grid md:grid-cols-2 md:gap-2"
              >
                <div
                  className="mr-1 xl:w-1/2 "
                >
                  <div
                    className="mb-2 block font-raleway"
                  >
                    <label
                      className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                      htmlFor="firstname"
                    >
                      First Name
                    </label>
                    <input
                      className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      id="firstName"
                      name="FirstName"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Ex: Yves"
                      type="text"
                    />
                  </div>
                  <p
                    className="text-red-800"
                  />
                </div>
                <div
                  className="ml-1 xl:w-1/2"
                >
                  <div
                    className="mb-2 block font-raleway"
                  >
                    <label
                      className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                      htmlFor="lastname"
                    >
                      Last Name
                    </label>
                  </div>
                  <input
                    className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    id="lastName"
                    name="LastName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Hakuzimana"
                    type="text"
                  />
                  <p
                    className="text-red-800"
                  />
                </div>
              </div>
              <div
                className="block mb-4 xl:flex md:grid md:grid-cols-2  md:gap-2"
              >
                <div
                  className="mr-1 xl:w-1/2"
                >
                  <div
                    className="mb-2 block font-raleway"
                  >
                    <label
                      className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                      htmlFor="email"
                    >
                      Email
                    </label>
                    <input
                      className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      id="email"
                      name="email"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Ex: example@provider.com"
                      type="email"
                    />
                  </div>
                  <p
                    className="text-red-800"
                  />
                </div>
                <div
                  className="ml-1 xl:w-1/2"
                >
                  <div
                    className="mb-2 block font-raleway"
                  >
                    <label
                      className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                      htmlFor="mobileNumber"
                    >
                      Mobile Number
                    </label>
                    <input
                      className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      id="mobileNumber"
                      name="mobileNumber"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="0789000000"
                      type="text"
                    />
                  </div>
                  <p
                    className="text-red-800"
                  />
                </div>
              </div>
              <div
                className="block mb-4 xl:flex md:grid md:grid-cols-2  md:gap-2"
              >
                <div
                  className="mr-1 xl:w-1/2"
                >
                  <label
                    className="block text-grey-darker text-sm font-bold mb-2"
                    htmlFor="company"
                    name="Company"
                  >
                    Company
                  </label>
                  <select
                    className="appearance-none border rounded w-full py-2 px-3 text-grey-darker bg-gray-200"
                    id="company"
                    placeholder="Select operator company"
                    required={true}
                  >
                    <option
                      className="text-grey-darker"
                      hidden={true}
                    >
                      Select company
                    </option>
                    <option
                      value=""
                    >
                      Royal Express Limited 
                    </option>
                    <option
                      value=""
                    >
                       City Express Limited 
                    </option>
                    <option
                      value=""
                    >
                      KIGALI BUS SERVICES
                    </option>
                  </select>
                </div>
                <div
                  className="ml-1 xl:w-1/2 "
                >
                  <div
                    className="mb-2 block font-raleway"
                  >
                    <label
                      className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                      htmlFor="address"
                    >
                      Address
                    </label>
                    <input
                      className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                      id="address"
                      name="address"
                      onBlur={[Function]}
                      onChange={[Function]}
                      placeholder="Cell, Sector, District"
                      type="text"
                    />
                  </div>
                  <p
                    className="text-red-800"
                  />
                </div>
              </div>
              <div
                className="mb-4"
              >
                <div
                  className="w-full mr-3 flex flex-col"
                >
                  <label
                    className="block text-grey-darker text-sm font-bold font-rale  mb-2"
                    htmlFor="nationalId"
                  >
                    National ID
                  </label>
                  <input
                    className="appearance-none border font-rale rounded w-full py-2 px-3 text-grey-darker bg-gray-200 text-md outline-hidden"
                    id="nationalId"
                    name="nationalid"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="1 1997 80020002 1 20"
                    type="text"
                  />
                  <p
                    className="text-red-800"
                  >
                     
                  </p>
                </div>
              </div>
              <div
                className="flex justify-center align"
              >
                <div>
                  <button
                    className="undefined bg-blue-500 hover:bg-blue-600 text-white font-bold py-2 px-4 rounded-xl m-2 "
                    type="submit"
                  >
                    Register Operator
                  </button>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
